handlers/changes.js

```diff
@@ -170,6 +170,11 @@ var getChanges = function(feed) {
     body: { doc_ids: _.union(feed.requestedIds, feed.validatedIds) },
     method: 'POST'
   }, function(err, changes) {
+    if (feed.res.finished) {
+      // don't write to the response if it has already ended
+      return;
+    }
+    cleanUp(feed);
```

Comment:
Yeah this is trying to clean up earlier to win the race condition! As discussed on Hangouts this shouldn't break anything...